generators/common/prompt-validators.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+var _ = require('lodash');
+var constants = require('./constants.js');
+var debug = require('debug')('generator-alfresco:prompt-validators');
+
+/**
+ * These validators are used to check the validity of the user input.
+ * We return a error message string when the input is invalid.
+ * If the input is valid, we return true
+ */
+
+module.exports = {
+
+  /**
+   * Given some input text and a array of validators,
+   * we sequentially pass the input to all validators unless it fails.
+   * Example: validators.sequentialValidatorFactory([validators.testValidator
+   *                            , validators.uniqueSourceAmpModuleValidator]),
+   *
+   * @param validatorArray
+   * @returns function
+   */
+  sequentialValidatorFactory: function (validatorArray) {
+    return function (input) {
+      return _.isEmpty(validatorArray) ? undefined : validatorArray.reduce(function (previousValue, validator) {
+        return (previousValue === true) ? validator.call(this, input) : previousValue;
+      }.bind(this), true);
+    };
+  },
+
+  uniqueSourceAmpModuleValidator: function (input, context) {
+    debug('uniqueSourceAmpModuleValidator(%s)', input);
+    if (_.isEmpty(input)) return 'Artifact Id cannot be empty';
+    var moduleRegistry = require('./alfresco-module-registry.js')(_.isEmpty(this.config) ? context : this);
```

Comment:
The 'context' argument is only used when running tests. Is this ok or is there a better way to handle it?